core/engine.go

```diff
@@ -57,23 +58,23 @@ type Engine struct {
 	ExecutionScheduler lib.ExecutionScheduler
 	executionState     *lib.ExecutionState
 
-	Options        lib.Options
+	options        lib.Options
 	runtimeOptions lib.RuntimeOptions
 	outputs        []output.Output
 
 	logger   *logrus.Entry
 	stopOnce sync.Once
 	stopChan chan struct{}
 
-	Metrics     map[string]*stats.Metric
+	Metrics     map[string]*stats.Metric // TODO: refactor, this doesn't need to be a map
```

Comment:
The To-do could benefit from mentioning what it could be instead 😸

Reply:
It's slightly better in the follow-up PR: https://github.com/grafana/k6/blob/60237cc7c3c15e93df91f4160dd4fd6ac3765a6c/metrics/engine/engine.go#L42-L48